ooni/reporter.py

```diff
@@ -298,8 +299,13 @@ def createReport(self):
             self.agent = SOCKS5Agent(reactor, proxyEndpoint=proxyEndpoint)
 
         elif self.collectorAddress.startswith('https://'):
-            # XXX add support for securely reporting to HTTPS collectors.
-            log.err("HTTPS based collectors are currently not supported.")
+            # not sure if there's something else it needs.  Seems to work.
+            # Very difficult to get it to work with self-signed certs.
+            self.agent = Agent(reactor)
```

Comment:
In testing, twisted 13.2 on python 2.7.6 didn't care about a non matching hostname or a self-signed certificate, and twisted 15.0 on python 2.7.3 rejects invalid hostnames and self-signed certificates, so I think we can infer that twisted 15.0 the preferable version.